jerry-core/ecma/operations/ecma-objects-general.c

```diff
@@ -658,77 +660,52 @@ ecma_op_general_object_define_own_property (ecma_object_t *obj_p, /**< the objec
   ecma_property_types_t current_type = ECMA_PROPERTY_GET_TYPE (current_p);
   const bool is_current_data_descriptor = (current_type == ECMA_PROPERTY_TYPE_NAMEDDATA);
   const bool is_current_accessor_descriptor = (current_type == ECMA_PROPERTY_TYPE_NAMEDACCESSOR);
+  const bool is_current_configurable = ecma_is_property_configurable (current_p);
 
   JERRY_ASSERT (is_current_data_descriptor || is_current_accessor_descriptor);
 
-  bool is_every_field_in_desc_also_occurs_in_current_desc_with_same_value = true;
-  if (property_desc_p->is_value_defined)
+  if (!is_current_configurable)
```

Comment:
I still have doubts here about equivalence with original code. Prove me wrong. If current is configurable, you sidestep these early return checks below and will certainly execute step 12, potentially with all the set and assign calls, even if unnecessary.